core/src/main/java/com/google/cloud/sql/core/InstanceConnectionNameResolver.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.core;
+
+/** Resolves the Cloud SQL Instance from the configuration name. */
+interface InstanceConnectionNameResolver {
```

Comment:
I think it's fine to make this package private for now -- but you'll want to plan for making this interface public so that callers can configure their own resolver.

Reply:
See https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/issues/1153 as an example of someone who already asked for this.